scmrepo/fs.py

```diff
@@ -160,6 +161,6 @@ def walk(  # pylint: disable=arguments-differ
 
     def ls(self, path, detail=True, **kwargs):
         for _, dirs, files in self.walk(path, detail=detail, **kwargs):
-            merge = files.update if detail else files.extend
-            merge(dirs)
-            return files
```

Comment:
Updating files return from walk in place is not as dangerous as as updating `dirs`(because it will change where walk proceeds walking into), but it is still not nice 🙂